taxonomy/management/commands/fetch_skill_details.py

```diff
@@ -0,0 +1,86 @@
+"""
+Management command for fetching and populating skill details, specifically skill category and subcategory.
+"""
+
+import logging
+from edx_django_utils.db import chunked_queryset
+
+from django.db.models import Q
+from django.core.management.base import BaseCommand, CommandError
+
+from taxonomy.emsi.client import EMSISkillsApiClient
+from taxonomy.emsi.parsers.skill_parsers import SkillDataParser
+from taxonomy.exceptions import TaxonomyAPIError
+from taxonomy.models import Skill, SkillCategory, SkillSubCategory
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command for fetching and populating skill details, specifically skill category and subcategory.
+
+    Example usage:
+        $ # Fetch skill category and subcategory for all the skills in the system.
+        $ ./manage.py fetch_skill_details
+        """
+    help = 'Fetch and populate skill category and subcategory.'
+
+    def _update_skill_category_and_sub_category(self, skill, skill_data):
+        """
+        Persist the skill category and subcategory data in the database.
+
+        Arguments:
+            skill (Skill): Skill instance whose category and subcategory needs to be added/updated.
+            skill_data (dict): A dictionary containing skill category and subcategory.
+        """
+        LOGGER.info('Updating category and subcategory data for skill external id {}.'.format(skill.external_id))
+        category = skill_data.get('category')
+        subcategory = skill_data.get('subcategory')
+        if category and subcategory:
```

Comment:
if catergory is there and only subcatergory is missing, do We want to save the category to the skill?

Reply:
it may be possible that there is no subcategory for s skill?

Reply:
Do we not want to save the category in case sub-category doesn't exist? If yes, why not?

Reply:
I have updated the logic to save category regardless of subcategory.